olp-cpp-sdk-dataservice-read/src/ReleaseDependencyResolver.cpp

```diff
@@ -47,55 +47,58 @@ const cache::KeyValueCache::KeyListType&
 ReleaseDependencyResolver::GetKeysToRelease(const TileKeys& tiles) {
   keys_to_release_.clear();
   for (const auto& tile : tiles) {
-    if (!ProcessTileKey(tile)) {
-      ProcessTileKeyInCache(tile);
-    }
+    ProcessTileKey(tile);
   }
   return keys_to_release_;
 }
 
-bool ReleaseDependencyResolver::ProcessTileKey(const geo::TileKey& tile_key) {
-  auto it = FindQuad(tile_key);
-  if (it != quad_trees_with_protected_tiles_.end()) {
-    // Quad tree for tile found. Get data handle for tile and add to list to
-    // release
-    auto tile_it = it->second.find(tile_key);
-    // if tile_key_to_release was not found, this means it is not
-    // protected
-    if (tile_it != it->second.end()) {
-      keys_to_release_.emplace_back(tile_it->second);
-      // key added, we can remove this tile from map
-      it->second.erase(tile_it);
-      if (it->second.empty()) {
-        // no more protected tiles associated with this quad tree
-        // can add key for quad tree to be released and remove from map
-        keys_to_release_.emplace_back(
-            partitions_cache_repository_.CreateQuadKey(
-                layer_id_, it->first, kQuadTreeDepth, version_));
-        quad_trees_with_protected_tiles_.erase(it);
+void ReleaseDependencyResolver::ProcessTileKey(const geo::TileKey& tile_key) {
+  bool add_key = true;
+  auto process_tile = [&](const geo::TileKey& quad_root,
+                          const geo::TileKey& tile_key) {
```

Comment:
Might be an issue with a duplicate name for tile_key. I would find another name, e.g. tile